src/esp32/pulse_counter.rs

```diff
@@ -0,0 +1,50 @@
+use esp_idf_sys::{pcnt_isr_service_install, pcnt_isr_service_uninstall, EspError, ESP_OK};
+use std::sync::atomic::{AtomicBool, AtomicU32, Ordering};
+use std::sync::Arc;
+
+/*
+This module exists because we want to ensure uniqueness of unit number
+across instances of an Esp32 Pulse Counter unit and make sure the isr service
+is only installed once.
+
+THIS MODULE IS A TEMPORARY MEASURE. When abstracting the atomicity of Esp32
+peripherals to board, this logic should be moved there.
+
+TODO: v5 of ESP-IDF has refactored pulse counter to manage what this module
+accomplishes for us. Potentially only use this module when on chips on v4.
+
+*/
+
+lazy_static::lazy_static! {
+    static ref NEXT_UNIT: Arc<AtomicU32> = Arc::new(AtomicU32::new(0));
+
+    static ref ISR_INSTALLED: Arc<AtomicBool> = Arc::new(AtomicBool::new(false));
```

Comment:
doesn't this count properly multiple instance of encoder (isr doesn't get uninstall when calling isr_uninstall?)

Reply:
I'm not sure what you mean here? If you mean that it appears dropping one encoder from memory will cause another to cease to function, this was an oversight. I'll change the behavior